app/assets/javascripts/spree/frontend/solidus_paypal_commerce_platform/buttons.js

```diff
@@ -2,58 +2,13 @@ SolidusPaypalCommercePlatform.renderButton = function(payment_method_id, style)
   paypal.Buttons({
     style: style,
     createOrder: function (data, actions) {
-      return Spree.ajax({
-        url: '/solidus_paypal_commerce_platform/orders',
-        method: 'POST',
-        data: {
-          payment_method_id: payment_method_id,
-          order_id: Spree.current_order_id,
-          order_token: Spree.current_order_token
-        }
-      }).then(function(res) {
-        return res.table.id;
-      })
+      return SolidusPaypalCommercePlatform.postOrder(payment_method_id);
```

Comment:
This and the following don't make any use of `this` in their implementation and can be used direclty as the value of the JS object, like in: `{…, createOrder: SolidusPaypalCommercePlatform.postOrder, … }`. 

Unless this is done intentionally to allow easier customization and look for the value at runtime 🤔

Reply:
I just tried this - PayPals JS checks whether createOrder initializes a function or not, and breaks if not. Unfortunately it looks like the surrounding function is necessary, unless I'm missing something.

![image](https://user-images.githubusercontent.com/5720486/85436029-feca4700-b54d-11ea-9424-c1a392ff611a.png)